include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -122,29 +121,21 @@ namespace internal
                                         Number>;
 
     static void
-    evaluate(const unsigned int                            n_components,
-             const EvaluationFlags::EvaluationFlags        evaluation_flag,
-             const MatrixFreeFunctions::ShapeInfo<Number> &shape_info,
-             const Number *                                values_dofs_actual,
-             Number *                                      values_quad,
-             Number *                                      gradients_quad,
-             Number *                                      hessians_quad,
-             Number *                                      scratch_data);
+    evaluate(const unsigned int                     n_components,
+             const EvaluationFlags::EvaluationFlags evaluation_flag,
```

Comment:
why not `evaluation_flags`?

Reply:
That's a good question, and I do not see a clear answer: We most often use `evaluation_flag`, but I found at least one occurrence of the opposite on the current master branch, https://github.com/dealii/dealii/blob/d6b0a8898f5eabb4fb7de369f8bc1e1c7134d07e/include/deal.II/matrix_free/fe_evaluation.h#L7238
I think the motivation to use `evaluation_flag` is to say that we have one specific option that should get evaluated in the end. So that could be interpreted as a single "flag" (in the sense of "setting") that gets passed around. However, the mask consists of multiple slots (booleans) to inform us about the combination of boolean flags to it is multiple options handled here, hence motivating the plural name `evaluation_flags`. Looking closer, this is also the grammar chosen by `UpdateFlags` in `FEValues`, e.g. https://github.com/dealii/dealii/blob/d6b0a8898f5eabb4fb7de369f8bc1e1c7134d07e/include/deal.II/fe/fe_values.h#L4053-L4060 , so I would lean towards the latter, also for consistency within the library. If I get some thumbs up here, I will change all appearances of `evaluation_flag` and `integration_flag` to the respective `_flags`.